lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -51,9 +52,19 @@ def virtual_total(name, relation, options = {})
       #    virtual_attribute :allocated_disk_storage, :integer, :uses => :disks, :arel => ...
       #
       #    # arel => (SELECT sum("disks"."size") where "hardware"."id" = "disks"."hardware_id")
+      %w[average maximum minimum].each do |m_name|
+        method = "virtual_#{m_name}"
+        define_method(method) do |name, relation, column, options = {}|
+          send("define_#{method}_method", name, relation, column)
+          send("define_virtual_aggregate_attribute", name, relation, m_name, column, options)
+        end
+      end
 
       def virtual_aggregate(name, relation, method_name = :sum, column = nil, options = {})
-        return define_virtual_total(name, relation, options) if method_name == :size
+        return virtual_total(name, relation, options) if method_name == :size
```

Comment:
so if method_name is any of these new three, @kbrock, we will never run `define_virtual_aggregate_method(name, relation, method_name, column)` which is what's checking to see if the data's been preloaded, i think, right?

Reply:
hmm, yea, lets sit on these for now but I'm definitely leaning towards skipping these and dumping this interface